### DISCUSSION TITLE
Debugging parcel plugin usage

### DISCUSSION THREAD

**Discussion by TLATER** [parcel-bundler/parcel#8493] [Q&A] [answered]
Hi, I'm doing something quite exotic, so in short:

How do I debug parcel's plugin usage, specifically why it is seemingly not using the posthtml transformer?

I'm looking for something like a log message stating which plugins parcel has available, and which it thinks it is configured to use for a specific file.

For context, I'm trying to use parcel in a nix build with [dream2nix](https://github.com/nix-community/dream2nix), which prepares a node env with parcel and a full `node_modules` directory, just without using npm to prepare any of that. My configuration works perfectly fine in a "shell"  environment, but as soon as I use a "build" environment parcel seems to not be using the posthtml transformer. This is strange, because the environments are supposed to be very similar.

Other transformers appear to work fine, e.g. the typescript transformer, so I think i'm hunting a weird environment bug, and I'm not sure whether this is in parcel or dream2nix, but either way I'm making no progress at all without any diagnostics, and `--log-level verbose` doesn't seem to be helping.

**Comment by mischnic** [comment, marked as answer]
Log messages like `Transforming node_modules/bootstrap/bootstrap.min.css with: "@parcel/transformer-postcss","@parcel/transformer-css"` would certainly be nice in some cases. But that doesn't exist (yet)

**Reply by TLATER** [reply]
Gotcha, thanks. Guess the best I can do is try to patch some debug statements into one of the plugins then.
